### PR TITLE
Use IndexHeading component on Most Read page

### LINKED ISSUE #6780: Use `IndexHeading` component on Most Read page
**Is your feature request related to a problem? Please describe.**
Once we a11y swarm the new `IndexHeading` component integrated in [this](https://github.com/bbc/simorgh/pull/6673) PR, we should also use it on the Most Read page.

**Describe the solution you'd like**
Replace the [MostReadHeader](https://github.com/bbc/simorgh/blob/latest/src/app/pages/MostReadPage/index.jsx#L52) styled component with the new `IndexHeading` component.

Please note that the paddings we use on the Most Read page are different from the ones set in the `IndexHeading` container, so this will have to be reviewed and adjusted to match the UX designs.

**Describe alternatives you've considered**
Leave it as it is.

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [X] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves #6780

**Overall change:**
Use the new `IndexHeading` component on the Most Read page.

**Code changes:**
- Bump `@bbc/psammead-heading-index` to the stable version
- Expand `IndexHeading` container to allow passing the `pageType` in order to apply the right paddings.
- Replace `MostReadHeader` with the `IndexHeading` in the Most Read page.

---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] I have added labels to this PR for the relevant pod(s) affected by these changes
- [X] I have assigned this PR to the Simorgh project

**Testing:**

- [X] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [X] This PR requires manual testing

### CODE REVIEW

File: src/app/pages/MostReadPage/index.jsx [hunk 1]
```diff
@@ -41,38 +36,20 @@ const ConstrainedWrapper = styled.div`
...
+
     dir,
     lang,
     mostRead: { header },
   } = useContext(ServiceContext);
 
   const MostReadWrapper = ({ children }) => (
     <ConstrainedWrapper>
-      <MostReadHeader script={script} service={service}>
+      <IndexHeading id="content" pageType="mostRead">
```
Comment (OlgaLyubin) [inline src/app/pages/MostReadPage/index.jsx:50]: Does this also need `tabIndex: '-1'` that was in Most Read's heading previously?

File: src/app/containers/IndexHeading/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -55,6 +55,116 @@ exports[`Index Heading snapshot should render correctly 1`] = `
...
+<h1
+  class="c0"
+  dir="ltr"
+  tabindex="-1"
+>
+  Index Heading
+</h1>
+`;
+
 exports[`Index Heading snapshot should render rtl correctly 1`] = `
```
Comment (OlgaLyubin) [inline src/app/containers/IndexHeading/__snapshots__/index.test.jsx.snap]: Seems like there are a couple of older snapshots still present in the file? There are 3 snapshot tests in the test file and 5 exports in the .snap file

File: package-lock.json [hunk 3]
```diff
@@ -14214,7 +14214,7 @@
         },
         "@jest/fake-timers": {
           "version": "24.9.0",
-          "resolved": "https://registry.npmjs.org/@jest/fake-timers/-/fake-timers-24.9.0.tgz",
+          "resolved": "https://npm.morph.int.tools.bbc.co.uk/@jest/fake-timers/-/fake-timers-24.9.0.tgz",
```
Comment (OlgaLyubin) [inline package-lock.json]: Can you please remove these changes?

File: src/app/containers/IndexHeading/index.jsx [hunk 4]
```diff
@@ -21,30 +27,57 @@ const StyledIndexHeading = styled(IndexHeading)`
...
+      {...props}
+    >
       {children}
     </StyledIndexHeading>
   );
 };
 
 IndexHeadingContainer.propTypes = {
   children: node,
+  pageType: PAGE_TYPES,
```
Comment (OlgaLyubin) [inline src/app/containers/IndexHeading/index.jsx]: Is there a reason why a definition for this is at the top of the page if `PAGE_TYPES` is only used for `propTypes`?

**Review by OlgaLyubin — COMMENTED** [review]
Looks good 👍  just a couple of comments

File: src/app/pages/MostReadPage/index.jsx (hunk 1 above)
Comment (DenisHdz) [reply src/app/pages/MostReadPage/index.jsx:50]: This is already applied inside the component in Psammead https://github.com/bbc/psammead/blob/latest/packages/components/psammead-heading-index/src/index.jsx#L9

File: src/app/containers/IndexHeading/index.jsx (hunk 4 above)
Comment (DenisHdz) [reply src/app/containers/IndexHeading/index.jsx]: It's fine either way. Just following the pattern we used in the `StoryPromo` container https://github.com/bbc/simorgh/blob/latest/src/app/containers/StoryPromo/index.jsx#L253

File: src/app/containers/IndexHeading/__snapshots__/index.test.jsx.snap (hunk 2 above)
Comment (DenisHdz) [reply src/app/containers/IndexHeading/__snapshots__/index.test.jsx.snap]: Weird 🤷🏻‍♂️. I've updated them

**Review by OlgaLyubin — APPROVED** [review]
Thanks for the changes, LGTM 👍

File: src/app/pages/IdxPage/index.jsx [hunk 5]
```diff
@@ -77,7 +77,9 @@ const IdxPage = ({
       <LinkedData type="WebPage" seoTitle={seoTitle} />
       <main role="main">
         <IndexPageContainer>
-          <IndexHeading id="content">{title}</IndexHeading>
+          <IndexHeading id="content" pageType="idx">
```
Comment (Bopchy) [inline src/app/pages/IdxPage/index.jsx:80]:
```suggestion
          <IndexHeading id="content">
```

Comment (Bopchy) [reply src/app/pages/IdxPage/index.jsx:80]: I think its safe to remove `pageType` here, as idx is set as the default in `IndexHeading`

**Review by Bopchy — APPROVED** [review]
Left a suggestion, otherwise LGTM. Thanks for the changes.

File: src/app/pages/IdxPage/index.jsx (hunk 5 above)
Comment (DenisHdz) [reply src/app/pages/IdxPage/index.jsx:80]: Agree, but I think it is better if we explicitly indicate it.

**Comment by PriyaKR** [comment]
LGTM.

**Pull request merged**